oitg/fitting/line.py

```diff
@@ -4,8 +4,8 @@
 from . import FitBase
 
 def parameter_initialiser(x, y, p):
-    k = (y[-1] - y[1]) / (x[-1] - x[1])
-    p['a'] = y[1] - x[1] * k
+    k = (y[-1] - y[0]) / (x[-1] - x[0])
+    p['a'] = y[0] - x[0] * k
```

Comment:
I would also accept ``np.max(y)- np.min(y) / (np.max(x) - np.min(x)``. The existing code assumes that the x & y arrays are sorted, not necessarily true.

Reply:
Good catch. None of the fitting routines should assume sorted data, so the max/min implementation is more appropriate.

Reply:
Another thought:
```python
p['b'] = (np.max(y) -np.min(y)) / (np.max(x) - np.min(x))
p['a'] = np.min(y) - np.min(x) * p['b']
```
IIRC this estimates the offset with the user specified gradient (if supplied). Not that it should matter much for fitting a line...